### ISSUE TITLE
Provide `AsScenarios` extension method

### ISSUE THREAD

**Issue by ehonda** [relaxdays/TestUtilities.Scenarios#81]
# Feature overview

A common usage pattern is to have code that looks as follows:

```csharp
// Convert a collection of values to scenarios by creating a description for each value
values.Select(value => value.AsScenario($"Some description for {value}"));
```

It would be nice to have a convenience method (which the inspiration for was actually hallucinated by GitHub Coplit 😉) that simplifies this use case, like this:

```csharp
values.AsScenarios(value => $"Some description for {value}");
```

# Usability

Since the extension method `AsScenario` (**singular**, i.e. the already existing method) is defined for any type, there's some slight concern it might be easy to accidentally use the wrong version:

```csharp
// ⚠ The value referred to in the lambda here is actually the whole collection and not an indivdual element
values.AsScenario(value =>  $"Some description for {value}");
```

So if there's a better name for the proposed `AsScenarios`, we could use it. On the other hand, it's kind of nice and consistent, and the error should become pretty obvious, because the resulting type (in case of confusion) will be e.g. `Scenario<IEnumerable<TElement>>`, instead of `IEnumerable<Scenario<TElement>>`, so there will probably be subsequent errors caused by this pointing to this root cause.

**Comment by Laxynium** [comment]
Hello, can I take this one as well?

**Comment by ehonda** [comment]
Sorry, I hadn't seen your comment. Yes, sure!

**Comment by ehonda** [comment]
Implemented in #90
